src/Illuminate/View/ViewServiceProvider.php

```diff
@@ -161,9 +161,14 @@ public function registerPhpEngine($resolver)
     public function registerBladeEngine($resolver)
     {
         $resolver->register('blade', function () {
-            $compiler = new CompilerEngine($this->app['blade.compiler'], $this->app['files']);
+            $app = app();
```

Comment:
@driesvints @nunomaduro 

As the `app()` helper is pulled from the Foundation component, doesn't the View component's `composer.json` file need to be updated to account for that?

Reply:
One option is to call `Container::getInstance()` as the `app()` helper basically does only that, and the Container component is already required by the view component.

For the other usage, `Container::getInstance()->make($dependency)` can be used.

https://github.com/laravel/framework/blob/bf7046f37a03ac006f60a811148f91393f305283/src/Illuminate/Foundation/helpers.php#L116-L123

Reply:
ping @taylorotwell